trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This file contains builders for GPflow models supported in Trieste. """
+
+from __future__ import annotations
+
+import math
+from typing import Optional
+
+import gpflow
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflow.models import GPR, SGPR, SVGP, VGP
+
+from ...data import Dataset
+from ...space import Box, SearchSpace
+from ...types import TensorType
+
+KERNEL_LENGTHSCALE = tf.cast(0.2, dtype=gpflow.default_float())
+"""
+Default value of the kernel lengthscale parameter.
+"""
+
+
+KERNEL_PRIOR_SCALE = tf.cast(1.0, dtype=gpflow.default_float())
+"""
+Default value of the scaling factor for the kernel lengthscale and variance parameters.
+"""
+
+
+CLASSIFICATION_KERNEL_VARIANCE_NOISE_FREE = tf.cast(100.0, dtype=gpflow.default_float())
+"""
+Default value of the kernel variance parameter for classification models in the noise free case.
+"""
+
+
+CLASSIFICATION_KERNEL_VARIANCE = tf.cast(1.0, dtype=gpflow.default_float())
+"""
+Default value of the kernel variance parameter for classification models.
+"""
+
+
+MAX_NUM_INDUCING_POINTS = tf.cast(500, dtype=tf.int32)
+"""
+Default maximum number of inducing points.
+"""
+
+
+NUM_INDUCING_POINTS_PER_DIM = tf.cast(25, dtype=tf.int32)
+"""
+Default number of inducing points per dimension of the search space.
+"""
+
+
+SNR_LIKELIHOOD = tf.cast(10, dtype=gpflow.default_float())
+"""
+Default value used for initializing (noise) variance parameter of the likelihood function.
+If user does not specify it, the noise variance is set to maintain the signal to noise ratio
+determined by this default value. Signal variance in the kernel is set to the empirical variance.
+"""
+
+
+def build_gpr(
+    data: Dataset,
+    search_space: SearchSpace,
+    kernel_priors: bool = True,
+    likelihood_variance: Optional[float] = None,
+    trainable_likelihood: bool = False,
+) -> GPR:
+    """
+    Build a :class:`~gpflow.models.GPR` model with sensible initial parameters and
+    priors. We use :class:`~gpflow.kernels.Matern52` kernel and
+    :class:`~gpflow.mean_functions.Constant` mean function in the model as we found this combination
+    to be effective in most settings.
```

Comment:
should we give a warning here (or on top of the page?) that this is a default configuration meant to work OK in most situation, but it should NOT be taken as a universally good solution?